contrib/gorilla/websocket/websocket_test.go

```diff
@@ -0,0 +1,158 @@
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_message", span.OperationName())
+				require.Equal(t, len("message"), span.Tag("websocket.message_length"))
+				require.Equal(t, websocket.TextMessage, span.Tag("websocket.message_type"))
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_message", span.OperationName())
+			},
+		},
+		{
+			name: "WritePrepareMessage",
+			client: func(conn *websockettrace.Conn) {
+				mt, msg, err := conn.ReadMessage()
+				require.NoError(t, err)
+				require.Equal(t, websocket.TextMessage, mt)
+				require.Equal(t, []byte("message"), msg)
+			},
+			server: func(conn *websockettrace.Conn) {
+				msg, err := websocket.NewPreparedMessage(websocket.TextMessage, []byte("message"))
+				err = conn.WritePreparedMessage(msg)
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_prepared_message", span.OperationName())
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_message", span.OperationName())
+			},
+		},
+		{
+			name: "WriteJSON/ReadJSON",
+			client: func(conn *websockettrace.Conn) {
+				var s string
+				err := conn.ReadJSON(&s)
+				require.NoError(t, err)
+				require.Equal(t, "message", s)
+			},
+			server: func(conn *websockettrace.Conn) {
+				err := conn.WriteJSON("message")
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_json", span.OperationName())
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_json", span.OperationName())
+			},
+		},
+		{
+			name:   "WriteControl",
+			client: func(*websockettrace.Conn) {},
+			server: func(conn *websockettrace.Conn) {
+				err := conn.WriteControl(websocket.PingMessage, []byte("message"), time.Now().Add(time.Second))
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 1)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_control", span.OperationName())
+				require.Equal(t, len("message"), span.Tag("websocket.message_length"))
+				require.Equal(t, websocket.PingMessage, span.Tag("websocket.message_type"))
+			},
+		},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			mt := mocktracer.Start()
+			defer mt.Stop()
+			runClientServer(t, tc.client, tc.server)
+			tc.assert(mt)
+		})
+	}
+}
+
+func runClientServer(t *testing.T, client, server func(conn *websockettrace.Conn)) {
```

Comment:
Sorry, but the fact that some code is a test helper, does not make it optional for testing. I think that in order to make sure our tests are accurate, it's mandatory to thoroughly test this function too, so I'd recommend having a `TestRunClientServer` bunch as well.

Reply:
But the test would fail by design if this function doesn't work 🤔
For example the span assertions would fail in such case.